bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/async/BigtableAsyncUtilities.java

```diff
@@ -42,142 +39,161 @@
+          ClientCall<RequestT, ResponseT> call = channel.newCall(method, CallOptions.DEFAULT);
+          addCancellationListener(cancellationToken, call);
+          CollectingClientCallListener<ResponseT> responseCollector =
+              new CollectingClientCallListener<>();
+          asyncServerStreamingCall(call, request, responseCollector);
+          return Futures.transform(responseCollector.getResponseCompleteFuture(), function);
         }
       };
+    }
 
-  public static BigtableAsyncRpc<SampleRowKeysRequest, List<SampleRowKeysResponse>>
-      createSampleRowKeyAsyncReader(Channel channel, ClientCallService clientCallService) {
-    return createStreamingAsyncRpc(channel, BigtableServiceGrpc.METHOD_SAMPLE_ROW_KEYS,
-      IMMUTABLE_LIST_TRANSFORMER, clientCallService);
-  }
-
-  public static BigtableAsyncRpc<ReadRowsRequest, List<Row>> createRowKeyAysncReader(Channel channel,
-      ClientCallService clientCallService) {
-    return createStreamingAsyncRpc(channel, BigtableServiceGrpc.METHOD_READ_ROWS, ROW_TRANSFORMER,
-      clientCallService);
-  }
-
-  public static <RequestT, ResponseT> BigtableAsyncRpc<RequestT, ResponseT> createAsyncUnaryRpc(
-      final Channel channel,
-      final ClientCallService clientCallService,
-      final MethodDescriptor<RequestT, ResponseT> method) {
-    return new BigtableAsyncRpc<RequestT, ResponseT>() {
-      @Override
-      public ListenableFuture<ResponseT> call(RequestT request, CancellationToken cancellationToken) {
-        ClientCall<RequestT, ResponseT> call = channel.newCall(method, CallOptions.DEFAULT);
-        addCancellationListener(cancellationToken, call);
-        return clientCallService.listenableAsyncCall(call, request);
-      }
-    };
-  }
-
-  private static <RequestT, ResponseT, OutputT>
-      BigtableAsyncRpc<RequestT, List<OutputT>> createStreamingAsyncRpc(
-          final Channel channel,
-          final MethodDescriptor<RequestT, ResponseT> method,
-          final Function<List<ResponseT>, List<OutputT>> function,
-          final ClientCallService clientCallService) {
-    return new BigtableAsyncRpc<RequestT, List<OutputT>>() {
-      @Override
-      public ListenableFuture<List<OutputT>> call(RequestT request, CancellationToken cancellationToken) {
-        ClientCall<RequestT, ResponseT> call = channel.newCall(method, CallOptions.DEFAULT);
-        addCancellationListener(cancellationToken, call);
-        CollectingStreamObserver<ResponseT> responseCollector = new CollectingStreamObserver<>();
-        clientCallService.asyncServerStreamingCall(call, request, responseCollector);
-        return Futures.transform(responseCollector.getResponseCompleteFuture(), function);
-      }
-    };
-  }
+    @Override
+    public <RequestT, ResponseT> void asyncServerStreamingCall(ClientCall<RequestT, ResponseT> call,
+        RequestT request, ClientCall.Listener<ResponseT> listener) {
+      call.start(listener, new Metadata());
+      call.request(1);
```

Comment:
Why requesting 1 here? A comment might be helpful, like it was above when unaryAsync requests 2.